corehq/util/timer.py

```diff
@@ -198,6 +198,26 @@ def to_list(self, exclude_root=False):
         """Get the list of ``NestableTimer`` objects in hierarchy order"""
         return self.root.to_list(exclude_root)
 
+    def add_to_sentry_breadcrumbs(self):
+        """Add timing context to the breadcrumbs section in sentry.
+
+        This must be called before sending to sentry, for instance if using
+        notify_exception rather than raising a hard error
```

Comment:
I believe this was the problem before - the case search view called `notify_exception`, and then _afterwards_, the middleware injected this context.

I also considered making `timing_context` an optional arg to `notify_exception`, so the order dependency can be handled there, but I wasn't sure if that was a good coupling.  You could in theory use this for requests that might 500 and not use `notify_exception` at all, but I suppose that's more theoretical.  Let me know your thoughts.

Reply:
I may not have the full picture but I think the theoretical scenario you highlighted sounds pretty reasonable? And therefore is worth keeping this separate from notify_exception/relying on the caller to order these correctly.